### PR TITLE
Support tag/module add/delete operations

### PR DESCRIPTION
**Pull request by salty-flower**
#69

### CODE REVIEW

File: src/main/java/seedu/connectus/logic/commands/AddTagToPersonCommand.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+public class AddTagToPersonCommand extends Command {
+    public static final String COMMAND_WORD = "addt";
+    public static final String MESSAGE_USAGE = ": Adds a %s to the person identified."
+        + "by the index number used in the displayed person list. \n"
+        + "Parameters: INDEX (must be a positive integer) "
+        + "[ " + PREFIX_MODULE + " MODULE]"
+        + "[ " + PREFIX_TAG + " TAG]"
+        + "\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_MODULE + "CS1231";
```
Comment (gremmyz) [inline src/main/java/seedu/connectus/logic/commands/AddTagToPersonCommand.java:33]: Minor details, should be

``` java
public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a tag to the person identified "
        + "by the index number used in the displayed person list. \n"
        + "Parameters: INDEX (must be a positive integer) "
        + "[" + PREFIX_MODULE + "MODULE] "
        + "[" + PREFIX_TAG + "TAG]"
        + "\n"
        + "Example: " + COMMAND_WORD + " 1 "
        + PREFIX_MODULE + "CS1231";
```

File: src/main/java/seedu/connectus/logic/commands/DeleteTagFromPersonCommand.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+public class DeleteTagFromPersonCommand extends Command {
+    public static final String COMMAND_WORD = "deletet";
+    public static final String MESSAGE_USAGE = ": Delete a tag from the person identified."
+        + "by the index number used in the displayed person list. \n"
+        + "Parameters: PERSON_INDEX (must be a positive integer) "
+        + "[ " + PREFIX_TAG + " TAG_INDEX]"
+        + "[ " + PREFIX_MODULE + " MODULE_INDEX]"
+        + "\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_MODULE + "1";
```
Comment (gremmyz) [inline src/main/java/seedu/connectus/logic/commands/DeleteTagFromPersonCommand.java:34]: Minor details, should be

``` java
    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Delete a tag from the person identified"
        + "by the index number used in the displayed person list. \n"
        + "Parameters: PERSON_INDEX (must be a positive integer) "
        + "[" + PREFIX_TAG + "TAG_INDEX]"
        + "[" + PREFIX_MODULE + "MODULE_INDEX]"
        + "\n"
        + "Example: " + COMMAND_WORD + " 1 "
        + PREFIX_MODULE + "1";
```

**Review by gremmyz — CHANGES_REQUESTED** [review]
I like this implementation, pretty cool O:

Some things to take note of:
- `addt 1`  and `deletet 1` should probably return an error (no prefixes), but it parses through.
- `addt 1 t/`, `addt 1 t/ ` (with whitespace), `addt 1 mod/`, `addt 1 mod/ ` (with whitespace) should return an error message, but error is only shown in the IDE console. No issue with `deletet` regarding this problem.

After that is resolved, I think it should be good to merge.

---
With these commands in place, do you guys think there's a need to depreciate tags in the `add` and `edit` function?

I'm thinking for `add`, we should keep it, but for `edit`, we should remove, and only use `addt` and `deletet` to edit a person's tags. What do you think?

---
**Notes for UG:**

`addt` command format: `addt INDEX [t/TAG] [mod/MODULE]`
`deletet` command format: `deletet PERSON_INDEX [t/TAG_INDEX] [mod/MODULE_INDEX]`

File: src/main/java/seedu/connectus/logic/commands/DeleteTagFromPersonCommand.java (hunk 2 above)
Comment (salty-flower) [reply src/main/java/seedu/connectus/logic/commands/DeleteTagFromPersonCommand.java:34]: Nice catch!

**Comment by salty-flower** [comment]
Regarding the fate of `edit`, I agree that we can retire its functionalities on `tag`s. It might be confusing for users to find their `tag`s wiped out after hitting `edit INDEX t/`.

**Review by gremmyz — APPROVED** [review]
Stuff from earlier review fixed. LGTM!

**Pull request merged**